test/data_base.cpp

```diff
@@ -1215,12 +1214,97 @@ BOOST_AUTO_TEST_CASE(data_base__update__new_transactions__success)
 
 // invalidate
 
-BOOST_AUTO_TEST_CASE(data_base__invalidate__missing_block__fails)
+#ifndef NDEBUG
+BOOST_AUTO_TEST_CASE(data_base__invalidate__missing_header__fails)
 {
+    create_directory(DIRECTORY);
+    bc::database::settings settings;
+    settings.directory = DIRECTORY;
+    settings.index_addresses = false;
+    settings.flush_writes = false;
+    settings.file_growth_rate = 42;
+    settings.block_table_buckets = 42;
+    settings.transaction_table_buckets = 42;
+    settings.address_table_buckets = 42;
+
+    data_base_accessor instance(settings);
+
+    static const auto bc_settings = bc::system::settings(bc::system::config::settings::mainnet);
+    BOOST_REQUIRE(instance.create(bc_settings.genesis_block));
+
+    const auto block1 = read_block(MAINNET_BLOCK1);
+
+    // setup ends
+    BOOST_REQUIRE_EQUAL(instance.invalidate(block1.header(), error::success), error::not_found);
 }
+#endif
 
-BOOST_AUTO_TEST_CASE(data_base__invalidate__existing__success)
+BOOST_AUTO_TEST_CASE(data_base__invalidate__validate__success)
 {
+    create_directory(DIRECTORY);
+    bc::database::settings settings;
+    settings.directory = DIRECTORY;
+    settings.index_addresses = false;
+    settings.flush_writes = false;
+    settings.file_growth_rate = 42;
+    settings.block_table_buckets = 42;
+    settings.transaction_table_buckets = 42;
+    settings.address_table_buckets = 42;
+
+    data_base_accessor instance(settings);
+
+    static const auto bc_settings = bc::system::settings(bc::system::config::settings::mainnet);
```

Comment:
Use a using for `bc::system::` (at least).

Reply:
Resolved for config::settings, but the other settings is available in both bc::system and bc::database. Could use typedef or `using = `, but might be too much?